smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/RuleSetParameterFinder.java

```diff
@@ -152,6 +153,64 @@ public Map<String, String> getContextParams(Shape operationInput) {
         return map;
     }
 
+    /**
+     * Get map of params to JavaScript equivalent of provided JMESPath expressions.
+     */
+    public Map<String, String> getOperationContextParamValues(OperationShape operation) {
+        Map<String, String> map = new HashMap<>();
+
+        Optional<OperationContextParamsTrait> trait = operation.getTrait(OperationContextParamsTrait.class);
+        if (trait.isPresent()) {
+            OperationContextParamsTrait operationContextParamsTrait = trait.get();
+            operationContextParamsTrait.getParameters().forEach((name, definition) -> {
+                String separator = ".";
+                String value = "this" + separator + "input";
+                String path = definition.getPath();
+
+                // Split JMESPath expression string on separator and add JavaScript equivalent.
+                for (String part : path.split("[" + separator + "]")) {
```

Comment:
is this string `"[.]"`?

Reply:
Yes. Java requires `.` to be escaped when padding to `.split(` function.

https://javarevisited.blogspot.com/2016/02/2-ways-to-split-string-with-dot-in-java-using-regular-expression.html